versioned/persist/tx/src/main/java/org/projectnessie/versioned/persist/tx/TxDatabaseAdapter.java

```diff
@@ -243,7 +248,20 @@ public Hash merge(
                     h -> {},
                     h -> {},
                     updateCommitMetadata);
-            return tryMoveNamedReference(conn, toBranch, currentHead, toHead);
+            Hash resultHash = tryMoveNamedReference(conn, toBranch, currentHead, toHead);
+            Hash parentId = getRefLogHead(conn);
+            Hash newRefLogId =
+                writeRefLogEntry(
+                    conn,
+                    toBranch.getName(),
+                    AdapterTypes.RefLogEntry.RefType.Branch,
+                    parentId,
+                    toHead,
```

Comment:
Is this correct?

Reply:
I think so as commitHash says the output hash of this operation. which is toHead. Anything I am missing ?